pkg/daemon/update.go

```diff
@@ -580,28 +573,19 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr err
 	}
 
 	// update files on disk that need updating
-	if err := dn.updateFiles(oldConfig, newConfig); err != nil {
+	if err := dn.updateFiles(oldIgnConfig, newIgnConfig); err != nil {
 		return err
 	}
 
 	defer func() {
 		if retErr != nil {
-			if err := dn.updateFiles(newConfig, oldConfig); err != nil {
+			if err := dn.updateFiles(oldIgnConfig, newIgnConfig); err != nil {
```

Comment:
any reason to swap oldConfig and newConfig param? defer function is supposed to get called when an error occurred in between and we would try to revert back the changes we applied.